webapp/tests/karma/ts/effects/reports.effects.spec.ts

```diff
@@ -885,6 +908,56 @@ describe('Reports effects', () => {
       }]);
     }));
 
+    it('should allow selecting a different report while completing verification', fakeAsync(() => {
+      const selectedReports = [{
+        _id: 'report',
+        doc: {
+          _id: 'report',
+          _rev: 2,
+          contact: { _id: 'contact', name: 'name', parent: { _id: 'parent', type: 'clinic' } },
+        },
+      }];
+      authService.has.resolves(true);
+      store.overrideSelector(Selectors.getSelectedReports, selectedReports);
+      store.refreshState();
+
+      sinon.stub(Date, 'now').returns(1000); // using faketimers breaks fakeAsync's tick :(
+      // Updating the report causes it to be re-selected
+      dbService.put.callsFake(() => {
+        actions$ = of(ReportActionList.selectReport({id: 'report', silent: true}));
```

Comment:
Since I have to fake this logic anyway, this test is technically not covering anything different than `should set report when a new report is selected while still loading an initial report` above. But it does more accurately reflect the workflow that caused the original issue (where verifying the report is what triggers the selection).  However, since this unit test does not hit the ReportsContentComponents logic for not re-selecting changed reports when the route id has changed, it does not truly test the verification race condition....

@dianabarsan let me know if you think I should just remove it because it is redundant, or if I should keep it because it illustrates the workflow....